scamp/spinn_srom.c

```diff
@@ -38,10 +38,11 @@ enum spinn_srom_commands {
     SROM_RDSR =     (0x05U << 24),
     SROM_WRSR =     (0x01U << 24),
     SROM_PE =       (0x42U << 24),
-    SROM_SE =       (0xd8U << 24),
-    SROM_CE =       (0xc7U << 24),
-    SROM_RDID =     (0xabU << 24),
-    SROM_DPD =      (0xb9U << 24)
+    // Out of range so shouldn't be used
+    // SROM_SE =       (0xd8U << 24),
```

Comment:
These should also be unsigned integers. See comment above about enums.

Reply:
Strangely, reenabling these doesn't seem to cause a warning any more...

Reply:
If you remove the pragma magic in spinnaker.h these warnings reappear.

I noticed the pragma __pop__ that should restore the warnings, but it doesn't seem to be having the expected effect.

A quick look around the ARM website suggests that there are scoping issues with pragmas, although I could only find function-related rules. Nothing related to enums.